File: lib/features/simulation/domain/models/simulation_entry.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:savvy/core/constants/financial_enums.dart';
import 'package:savvy/features/simulation/domain/models/simulation_change.dart';

part 'simulation_entry.freezed.dart';
part 'simulation_entry.g.dart';

@freezed
abstract class SimulationEntry with _$SimulationEntry {
  const SimulationEntry._();

  const factory SimulationEntry({
    required String id,
    required String title,
    String? description,
    SimulationTemplate? template,
    @Default('sparkles') String iconName,
    @Default('#3F83F8') String colorHex,
    @Default([]) List<SimulationChange> changes,
    String? compareWithId,
    // Legacy fields — kept for backward compat with old Firestore data
    // ignore: deprecated_member_use_from_same_package
    SimulationType? type,
    @Default({}) Map<String, dynamic> parameters,
    @Default(false) bool isIncluded,
    @Default(false) bool isDeleted,
    required DateTime createdAt,
    DateTime? updatedAt,
  }) = _SimulationEntry;

  factory SimulationEntry.fromJson(Map<String, dynamic> json) =>
      _$SimulationEntryFromJson(json);
}

File: SavvyApp/Packages/Core/SavvyFoundation/Sources/SavvyFoundation/Models/SimulationEntry.swift
import Foundation

public struct SimulationEntry: FirestoreEntity {
    public let id: String
    public var title: String
    public var description: String?
    public var template: SimulationTemplate?
    public var iconName: String
    public var colorHex: String
    public var changes: [SimulationChange]
    public var compareWithId: String?
    public var isIncluded: Bool
    public var isDeleted: Bool
    public let createdAt: Date
    public var updatedAt: Date?

    public init(
        id: String = UUID().uuidString,
        title: String,
        description: String? = nil,
        template: SimulationTemplate? = nil,
        iconName: String = "sparkles",
        colorHex: String = "#3F83F8",
        changes: [SimulationChange] = [],
        compareWithId: String? = nil,
        isIncluded: Bool = false,
        isDeleted: Bool = false,
        createdAt: Date = Date(),
        updatedAt: Date? = nil
    ) {
        self.id = id
        self.title = title
        self.description = description
        self.template = template
        self.iconName = iconName
        self.colorHex = colorHex
        self.changes = changes
        self.compareWithId = compareWithId
        self.isIncluded = isIncluded
        self.isDeleted = isDeleted
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}
